migration/identity-migration/src/main/java/io/camunda/migration/identity/config/MigrationHandlerConfig.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.migration.identity.config;
+
+import io.camunda.migration.identity.GroupMigrationHandler;
+import io.camunda.migration.identity.midentity.ManagementIdentityClient;
+import io.camunda.security.auth.Authentication;
+import io.camunda.service.GroupServices;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnCloud
+public class MigrationHandlerConfig {
```

Comment:
I went for a dedicated explicit config class to list all handlers per mode, I personally think this is easier to get an overview which are active compared to having them all annotated as Component and spreading conditions all over them. Wdyt?

Reply:
> That means that we have to create 3 config classes right ? O

Yes we could also have them as nested classes to see the different setups at one glance.

>  Do you also want to create specific handlers for each mode ?

I think there will be cases where a dedicated handler might be the simplest, as they differ completely, like the SaaSRoles one? For others we might just encapsulate different "sources" of the data by some fetch-service?